File: GestionEvenementFront/src/app/_Models/Logistique.ts
import {Ordre} from "./Ordre";
import {Payement} from "./Payement";
import {Event} from "./Event";

export class Logistique {
  logistiqueId:string;
  depenses:number;
  event:Event;
  payement:Payement ;
  ordres:Ordre[] ;
  totalDepenses:number;
  totalIncome:number;
}

File: GestionEvenementBack/src/main/java/tn/eesprit/gestionevenementback/Entities/Logistique.java
package tn.eesprit.gestionevenementback.Entities;

import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;
import java.util.List;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class Logistique implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer logistiqueId;
    Float depenses;
    Float totalDepenses;
    Float totalIncome;


    @JsonIgnore
    @OneToOne(cascade = CascadeType.ALL)
    Event event;

    @OneToOne(mappedBy = "logistique",cascade = CascadeType.REMOVE)
    Payement payement;
    @JsonIgnore
    @OneToMany(mappedBy = "logistique")
    Set<Ordre> ordres;
}
